local_planner/src/nodes/common.cpp

```diff
@@ -13,26 +13,14 @@ float distance2DPolar(int e1, int z1, int e2, int z2) {
   return sqrt(pow((e1 - e2), 2) + pow((z1 - z2), 2));
 }
 
-float computeL2Dist(const geometry_msgs::Point& position,
-                    const pcl::PointXYZ& xyz) {
-  return sqrt(pow(position.x - xyz.x, 2) + pow(position.y - xyz.y, 2) +
-              pow(position.z - xyz.z, 2));
-}
-
-float distance3DCartesian(const geometry_msgs::Point& a,
-                          const geometry_msgs::Point& b) {
-  return sqrt((a.x - b.x) * (a.x - b.x) + (a.y - b.y) * (a.y - b.y) +
-              (a.z - b.z) * (a.z - b.z));
-}
-
 // transform polar coordinates into Cartesian coordinates
-geometry_msgs::Point fromPolarToCartesian(int e, int z, double radius,
-                                          const geometry_msgs::Point& pos) {
-  geometry_msgs::Point p;
-  p.x =
+Eigen::Vector3f fromPolarToCartesian(int e, int z, double radius,
+                                     const geometry_msgs::Point& pos) {
+  Eigen::Vector3f p;
+  p.x() =
       pos.x + radius * cos(e * (M_PI / 180)) * sin(z * (M_PI / 180));  // round
-  p.y = pos.y + radius * cos(e * (M_PI / 180)) * cos(z * (M_PI / 180));
-  p.z = pos.z + radius * sin(e * (M_PI / 180));
+  p.y() = pos.y + radius * cos(e * (M_PI / 180)) * cos(z * (M_PI / 180));
+  p.z() = pos.z + radius * sin(e * (M_PI / 180));
```

Comment:
Lots of implicit type conversions here. Mixing int, float and double... Since you seem to have chosen Vector3f, maybe single-precision floating points all around?

Reply:
Agreed, after the Eigen conversion everything should be floats.